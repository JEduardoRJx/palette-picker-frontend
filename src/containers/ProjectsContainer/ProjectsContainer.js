import React, { Component } from 'react';
import './ProjectsContainer.scss';
import trash from '../../images/trash.svg'


export const ProjectsContainer = ({ deleteProject, deletePalette, projects, palettes }) => {
  console.log('2', palettes)
  const displayPalettes = palettes.map(palette => {
    return (
      <div className='palette' key={palette[0].id}>
          <div className='palette-name-wraper' id={`${palette.id}`}>
            <h3>{palette[0].palette_name}</h3>
            <button className='delete-btn palette-trash' 
              type='button' 
              id={`${palette[0].id}`}
              onClick={(e) => deletePalette(e)}>
            <img className='delete-icon palette-trash' src={trash} id={`${palette[0].id}`} />
            </button>
          </div>
          <div className='color-wrapper'>
            <div className='color' style={{backgroundColor: palette[0].color1}}></div>
            <div className='color' style={{backgroundColor: palette[0].color2}}></div>
            <div className='color' style={{backgroundColor: palette[0].color3}}></div>
            <div className='color' style={{backgroundColor: palette[0].color4}}></div>
            <div className='color' style={{backgroundColor: palette[0].color5}}></div>
          </div>
      </div>
    )
  })

  const displayProjects = projects.map(project => {
    return (
      <div className='individual-project project' id={project.id} key={project.id}>
        <div className='project-name-wrapper'>
          <h2 className='project-name'>{project.project_name}</h2>
          <button className='delete-btn project-trash' 
            type='button' 
            id={`${project.id}`}
            onClick={(e) => deleteProject(e)}>
          <img className='delete-icon project-trash' src={trash} id={project.id} />
          </button>
        </div>
          {displayPalettes}
      </div>
    )
  })

  return (
    <section className='projects-container'>
      {displayProjects}
    </section>
  )
}