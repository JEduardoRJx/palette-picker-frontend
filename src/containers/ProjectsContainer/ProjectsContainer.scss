.projects-container {
  height: 100vh;
  width: 100%;
  // background-color: blue;
}

.project-name {
  margin-top: 2%;
  font-size: 2em;
}

.projects-wrapper {
  height: 150px;
  width: 90%;
  border: 5px solid;
  border-radius: 5px;
  margin: 0 auto;
}

.palettes-wrapper {
  height: 35%;
  width: 90%;
  border: 5px solid;
  border-radius: 5px;
  margin: 0 auto;
  margin-top: 2%;
  padding: 10px;
}

.palette-name {
  font-size: 2em;
}

.delete-btn {
  width: 10%;
  height: 100%;
  // margin-top: 10%;
  // background-color: blue;
  border: none;
  outline: none;
  border-radius: 10px;
  padding: 3px;
  &:hover {
    background-color: rgb(243, 171, 171);
  }
}

.delete-icon {
  width: 100%;
  height: 100%;
}

.palette {
  // background-color: rgb(71, 128, 71);
  width: 45%;
  margin-bottom: 1%;
  text-align: center;
}

.name-delete-wrapper {
  width: 100%;
  height: 50px;
  // background-color: purple;
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-bottom: 1%;
}

ul {
  display: flex;
  justify-content: space-around;
}

.color {
  height: 100px;
  width: 100px;
  background-color: lightblue;
  border-radius: 50%;
}