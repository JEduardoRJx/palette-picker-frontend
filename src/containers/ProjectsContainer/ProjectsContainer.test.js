import React from 'react';
import { shallow } from 'enzyme';
import { ProjectsContainer } from './ProjectsContainer';

describe('ProjectsContainer ', () => {
  const mockPalettes = [
    [{
      id: 3,
      palette_name: "spring colors",
      project_id: 3,
      color1: "#CECCCC",
      color2: "#9D6381",
      color3: "#FDECEF",
      color4: "#0F110C",
      color5: "#612940",
      created_at: "2019-12-10T04:14:29.641Z",
      updated_at: "2019-12-10T04:14:29.641Z"
    }]
  ];

  const mockProjects = [
    {
      id: 3,
      project_name: "spring",
      user_id: 1,
      created_at: "2019-12-10T04:14:29.628Z",
      updated_at: "2019-12-10T04:14:29.628Z"
    }
  ]

  it('should match the snapshot ', () => {
    const wrapper = shallow(<ProjectsContainer 
      palettes={mockPalettes}
      projects={mockProjects}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});