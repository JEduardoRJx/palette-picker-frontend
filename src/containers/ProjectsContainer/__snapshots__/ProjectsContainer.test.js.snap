// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectsContainer  should match the snapshot  1`] = `
<section
  className="projects-container"
>
  <div
    className="individual-project project"
    id={3}
    key="3"
  >
    <div
      className="project-name-wrapper"
    >
      <h2
        className="project-name"
      >
        spring
      </h2>
      <button
        className="delete-btn project-trash"
        id="3"
        onClick={[Function]}
        type="button"
      >
        <img
          className="delete-icon project-trash"
          id={3}
          src="trash.svg"
        />
      </button>
    </div>
    <div
      className="palette"
      key="3"
    >
      <div
        className="palette-name-wraper"
        id="undefined"
      >
        <h3>
          spring colors
        </h3>
        <button
          className="delete-btn palette-trash"
          id="3"
          onClick={[Function]}
          type="button"
        >
          <img
            className="delete-icon palette-trash"
            id="3"
            src="trash.svg"
          />
        </button>
      </div>
      <div
        className="color-wrapper"
      >
        <div
          className="color"
          style={
            Object {
              "backgroundColor": "#CECCCC",
            }
          }
        />
        <div
          className="color"
          style={
            Object {
              "backgroundColor": "#9D6381",
            }
          }
        />
        <div
          className="color"
          style={
            Object {
              "backgroundColor": "#FDECEF",
            }
          }
        />
        <div
          className="color"
          style={
            Object {
              "backgroundColor": "#0F110C",
            }
          }
        />
        <div
          className="color"
          style={
            Object {
              "backgroundColor": "#612940",
            }
          }
        />
      </div>
    </div>
  </div>
</section>
`;
